src/App.tsx

```diff
@@ -1,57 +1,38 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Autocomplete } from './Autocomplete';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const handleSelectedPerson = (person: Person | null) => {
+    setSelectedPerson(person);
+  };
+
+  const headerText = selectedPerson
+    ? `${selectedPerson.name} (${selectedPerson.born} = ${selectedPerson.died})`
+    : 'No selected person';
+
+  useEffect(() => {
+    if (!selectedPerson) {
+      handleSelectedPerson(null);
+    }
+  }, [selectedPerson]);
 
```

Comment:
this useEffect doesn't make sense - if `selectedPerson === null` (it could be null or Person object) you are trying to set it `null` again

remove it